react/tic-tac-toe/src/app/screens/Game/components/MatchHistory/index.js

```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react';
+import styles from './styles.module.scss';
+import matches from "~services/MatchesService";
+import { connect } from 'react-redux';
+import matchActions from "~redux/matches/actions";
+import Spinner from 'react-spinkit';
+import { arrayOf, func, boolean } from 'prop-types';
+import { matchPropType } from "~constants/propTypes.js"
+import {PLAYER_ONE, PLAYER_TWO, WINNER, MATCH_HISTORY_TITLE, GET_ERROR} from "~constants";
+
+class MatchHistory extends Component {
+
+    componentDidMount = () => {
+      const { toggleLoading, getMatches} = this.props;
+      toggleLoading();
+      matches.getMatches().then(response => getMatches(response.data));
+    }
+
+    renderLine = data => {
+      return (
+      <div key={data.id}>
+        {PLAYER_ONE} {data.player_one}, {PLAYER_TWO} {data.player_two}, {WINNER} {data.winner}
+      </div>
+    )}
+
+    render() {
+        const { hasError, isLoading, matchesHistory} = this.props;
+        const matchesLines = matchesHistory ? matchesHistory.map(this.renderLine) : GET_ERROR;
```

Comment:
You can write hasError || !matches ? GET_ERROR : matchesHistory.map(this.renderLine) and then you don't need add a new ternary